ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -552,10 +552,51 @@ public func mergeMap<T, U>(transform: T -> SignalProducer<U>)(producer: SignalPr
 public func switch<T>(producer: SignalProducer<SignalProducer<T>>) -> SignalProducer<T>
 public func switchMap<T, U>(transform: T -> SignalProducer<U>)(producer: SignalProducer<T>) -> SignalProducer<U>
 
-public func repeat<T>(count: Int)(producer: SignalProducer<T>) -> SignalProducer<T>
 public func retry<T>(count: Int)(producer: SignalProducer<T>) -> SignalProducer<T>
 */
 
+/// Repeats `producer` a total of `count` times.
+/// Repeating `1` times results in a equivalent signal producer.
+public func times<T, E>(count: Int)(producer: SignalProducer<T, E>) -> SignalProducer<T, E> {
+	precondition(count >= 0)
+
+	if count == 0 {
+		return .empty
+	} else if count == 1 {
+		return producer
+	}
+
+	return SignalProducer { observer, disposable in
+		let serialDisposable = SerialDisposable()
+		disposable.addDisposable(serialDisposable)
+
+		var remainingTimes = count
+
+		// Allows recursive call.
+		// Radar: http://openradar.me/19861074
+		var iterate: (() -> ())!
+		iterate = {
+			producer.startWithSignal { signal, signalDisposable in
+				serialDisposable.innerDisposable = signalDisposable
+
+				signal.observe(next: { value in
+					sendNext(observer, value)
+				}, error: { error in
+					sendError(observer, error)
+				}, completed: {
+					if --remainingTimes > 0 {
+						iterate()
```

Comment:
I think this will strongly capture the block within itself, creating a leak.

You might want @andymatuschak's [implementation of the Z combinator](https://github.com/ReactiveCocoa/ReactiveCocoa/blob/63f55928b5cc613433bfe252cc66ee5ce69df8bd/ReactiveCocoa/Swift/SignalProducer.swift#L535-L539) instead.